Add support to build without librtlsdr and use one Windows UWP

AlwinEsch opened the pull request:
Not runtime tested now on Windows UWP.
I will look in next time to become the Kodi itself with Win UWP working on my system.

AlwinEsch committed:
fix sqlite3 depends build within Windows UWP

AlwinEsch committed:
update depends libusb about Windows to version 1.0.27

AlwinEsch committed:
remove wrongly added CMakeLists.txt.orig

AlwinEsch commented:
This change mainly about the source code only, with the settings.xml need a new feature added in Kodi to prevent show of USB settings on Windows UWP system (Currently only about Apple OS parts in Kodi possible).

AlwinEsch committed:
fix pthread-win32 depends build within Windows UWP

AlwinEsch committed:
don't use pthread-win32 on Windows UWP

AlwinEsch committed:
add another PARITY(x) for Windows UWP ARM builds within dsp_hd

As there fails the Windows UWP ARM build implements this a own function as fallback to use.

AlwinEsch committed:
fix linux distribution build (Ubuntu) by allow -DDISTRIBUTION_BUILD=1

This adds the cmake option "-DDISTRIBUTION_BUILD=1" to inform add-on build
about Linux Distribution build.
This becomes needed as on Ubuntu used libusb becomes linked static, but not created
as "position independent code". With the option becomes the linkage about prevented,
as the lib already as dependency in librtlsdr becomes it to add-on as shared on this
way added.

ksooo commented on src/dsp_hd/conv_dec.c:
```diff
@@ -19,6 +19,16 @@
+// This fix prevent the use about if something has set in defines.
+//
+#if defined(HAVE_SSE3) && defined(TARGET_LINUX)
+#warning SSE3 use defined, due to compile problems about, it becomes undefined for use!
```
Maybe this can be rephrased a bit: "Undefined HAVE_SSE3 to workaround compile problems."

ksooo commented on src/exception_control/win32_exception.cpp:
```diff
@@ -108,9 +108,15 @@ std::string win32_exception::format_message(DWORD result)
+#ifndef TARGET_WINDOWS_STORE
   DWORD cchformatted = FormatMessageA(
       FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
       nullptr, result, GetThreadUILanguage(), reinterpret_cast<LPSTR>(&formatted), 0, nullptr);
```
Could the `GetThreadUILanguage()`also be done for desktop windows? Then we could get rid of the `#ifdef`

AlwinEsch replied:
Have set for both to 0 and the #ifndef removed.
Also on the other comment I changed the warning text.

AlwinEsch committed:
add hack fix to not use SSE3 on Linux

It produced before errors e.g. like this:
/home/alwin/Dev/addons/pvr/pvr.rtlradio/src/dsp_hd/conv_sse.h:283:9: note: in expansion of macro ‘SSE_BRANCH_METRIC_N4’
  283 |         SSE_BRANCH_METRIC_N4(m0, m1, m2, m3, m7, m7)
      |         ^~~~~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-linux-gnu/13/include/tmmintrin.h:52:1: error: inlining failed in call to ‘always_inline’ ‘_mm_hadds_epi16’: target specific option mismatch
   52 | _mm_hadds_epi16 (__m128i __X, __m128i __Y)

Tried some ways to fix in code, but no matter which try it always fails somewhere.

AlwinEsch committed:
allow build without librtlsdr usage and only use network

ksooo reviewed (approved):
Great to see all builds green now. Thanks Alwin.
